### PR TITLE
Added radix sort and Shell sort 

### LINKED ISSUE #1909: [Idea]:  Shell Sort
### Idea Title

Add a new sorting algorithm

### Idea Description

An optimized version of insertion sort. Shell Sort sorts elements far apart first, then progressively reduces the gap.

@ajay-dhangar can you assign this to me??

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [X] Improves code quality
- [ ] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by laxmikandivalasa**
### Description
Added radix sort and Shell sort

Fixes #1909 #1908 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X ] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my code
- [X ] I have commented my code, particularly in hard-to-understand areas
- [ X] I have added tests that prove my fix is effective or that my feature works
- [ X] New and existing unit tests pass locally with my changes
- [ X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by laxmikandivalasa** [comment]
@ajay-dhangar you can check the new algos added
can you assign me with level2 since i have resolved both the issues

File: docs/sortings/radix-sort.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        }
+    }
+
+    public static void main(String[] args) {
+        int[] arr = {170, 45, 75, 90, 802, 24, 2, 66};
+        radixSort(arr);
+
+        System.out.println("Sorted array: " + Arrays.toString(arr));
+    }
+}
```
Comment (ajay-dhangar) [inline docs/sortings/radix-sort.md]: close your code example with ```` ``` ````

File: docs/sortings/shell-sort .md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        }
+    }
+
+    public static void main(String[] args) {
+        int[] arr = {5, 2, 9, 1, 5, 6};
+        shellSort(arr);
+
+        System.out.println("Sorted array: " + Arrays.toString(arr));
+    }
+}
```
Comment (ajay-dhangar) [inline docs/sortings/shell-sort .md]: close your code example with ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/sortings/shell-sort .md (hunk 2 above)
Comment (laxmikandivalasa) [reply docs/sortings/shell-sort .md]: sure!

**Comment by laxmikandivalasa** [comment]
@ajay-dhangar you can check now!!

**Pull request closed without merging**